components/FirebaseContext/index.tsx

```diff
@@ -0,0 +1,151 @@
+import {
+  useState,
+  createContext,
+  useContext,
+  useCallback,
+  useEffect,
+  PropsWithChildren,
+  SetStateAction
+} from 'react'
+import { User, Auth, signOut } from 'firebase/auth'
+import {
+  doc,
+  Firestore,
+  setDoc,
+  updateDoc,
+  FirestoreError
+} from 'firebase/firestore'
+import { useAuthState } from 'react-firebase-hooks/auth'
+import FireStoreParser from 'firestore-parser'
+import { auth, db } from 'pages/api/firebase'
+import { MESSAGES } from '@components/Auth/enums'
+
+const FIRESTORE_URL =
+  'https://firestore.googleapis.com/v1/projects/foodbank-c9a2f/databases/(default)/documents/users'
+
+interface FirebaseContextProps {
+  auth: Auth
+  db: Firestore
+  user?: User | null
+  userLoading?: boolean
+  userError?: Error
+  achievementsCount: AchievementsCountProp
+  addAchievementsCount?: (newAchievementsEarned: number) => void
+  signOutClearDataUnlockGrid?: () => void
+  gridDisabled?: boolean
+  setGridDisabled?: (value: SetStateAction<boolean>) => void
+}
+
+interface AchievementsCountProp {
+  count: number
+}
+const defaultAchievementsCount: AchievementsCountProp = { count: 0 }
+
+const FirebaseContext = createContext<FirebaseContextProps>({
+  auth: auth,
+  db: db,
+  achievementsCount: defaultAchievementsCount
+})
+const useFirebase = () => useContext(FirebaseContext)
+
+const FirebaseProvider = ({ children }: PropsWithChildren<{}>) => {
```

Comment:
Seems unnecessary to declare `PropsWithChildren` is you have no other props.
```suggestion
const FirebaseProvider: FC = ({ children }) => {
```
Make sure to import the required fields